[export] Add unit test for SDPA export result

larryliu0820 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #117390

Summary:

A follow up for #117097. In that PR I didn't add
`_scaled_dot_product_attention_for_cpu` into the core_aten_decomposition
table. This PR does that and also add a unit test.

Test Plan: python test/export/test_export.py -k
test_scaled_dot_product_attention

Reviewers:

Subscribers:

Tasks:

Tags:

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

larryliu0820 committed:
[export] Add unit test for SDPA export result

Summary:

A follow up for #117097. In that PR I didn't add
`_scaled_dot_product_attention_for_cpu` into the core_aten_decomposition
table. This PR does that and also add a unit test.

Test Plan: python test/export/test_export.py -k
test_scaled_dot_product_attention

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

drisspg commented on test/export/test_export.py:
```diff
@@ -2528,6 +2529,41 @@ def forward(self, x):
+        torch/_decomp/decompositions.py along with the kernel changes, so all of the
+        downstream backends are not being affected.
+        """
+        class ScaledDotProductAttention(torch.nn.Module):
```
Can we also add a test for "cuda" flow, or does that not matter for export?

For instance I imagine that: https://github.com/pytorch/pytorch/pull/115663, would break export of sdpa for gpu.
And that the fix would look like: https://github.com/pytorch/pytorch/pull/117097

I would hope that this would error on CI/CD and then we can fix in the PR

larryliu0820 replied:
Added

larryliu0820 committed:
Update on "[export] Add unit test for SDPA export result"


Summary:

A follow up for #117097. In that PR I didn't add
`_scaled_dot_product_attention_for_cpu` into the core_aten_decomposition
table. This PR does that and also add a unit test.

Test Plan: python test/export/test_export.py -k
test_scaled_dot_product_attention

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

drisspg reviewed (commented):
I think the tests look good, but the failures in the hud seem real

larryliu0820 committed:
Update on "[export] Add unit test for SDPA export result"


Summary:

A follow up for #117097. In that PR I didn't add
`_scaled_dot_product_attention_for_cpu` into the core_aten_decomposition
table. This PR does that and also add a unit test.

Test Plan: python test/export/test_export.py -k
test_scaled_dot_product_attention

Reviewers:

Subscribers:

Tasks:

Tags:

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

larryliu0820 commented:
> I think the tests look good, but the failures in the hud seem real

Fixed

larryliu0820 commented:
@pytorchbot merge -f "Error seems unrelated"
